### PR TITLE
feat: UX: Multichain: Show 'connect account' toast when current account isn't connected

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

When the user opens MetaMask on a dApp where the current account is *not* connected but another is, show a toast to the user asking if they'd like to connect this account.

## **Related issues**

Fixes:https://app.zenhub.com/workspaces/metamask-wallet-ux-63529dce65cbb100265a3842/issues/gh/metamask/metamask-planning/1977

## **Manual testing steps**

1.  Select Account 1
2. Go to Uniswap, connect to the dApp
3. Switch to Account 2
4. Open MetaMask, see the connection toast
5. Click to connect; see toast disappear and connection tooltip display

### **Before**

N/A

### **After**

https://github.com/MetaMask/metamask-extension/assets/46655/132ced9e-f90f-42a0-a63f-4d595f4c593d

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by vthomas13 — DISMISSED** [review]

**Comment by darkwing** [comment]
Started an E2E but will need to be a follow up PR -- getting it working properly is difficult at the moment because we don't have a way to open the extension on a dApp without the action closing the popup right away.

```
const { strict: assert } = require('assert');
const { toHex } = require('@metamask/controller-utils');
const {
  defaultGanacheOptions,
  withFixtures,
  openDapp,
  DAPP_URL,
  DAPP_ONE_URL,
  unlockWallet,
  WINDOW_TITLES,
  generateGanacheOptions,
  regularDelayMs,
  switchToNotificationWindow,
} = require('../helpers');
const FixtureBuilder = require('../fixture-builder');
const { PAGES } = require('../webdriver/driver');

describe('Dapp connect toast', function () {
  it('should display when an account other than the current account is connected', async function () {
    await withFixtures(
      {
        dapp: true,
        fixtures: new FixtureBuilder().build(),
        ganacheOptions: generateGanacheOptions({
          concurrent: { port: 8546, chainId: 1338 },
        }),
        title: this.test.fullTitle(),
      },
      async ({ driver }) => {
        await driver.navigate();

        await unlockWallet(driver);
        await openDapp(driver);

        // Connect the dapp
        await driver.clickElement('#connectButton');
        await driver.delay(regularDelayMs);
        await driver.waitUntilXWindowHandles(3);
        await switchToNotificationWindow(driver);
        await driver.clickElement({
          text: 'Next',
          tag: 'button',
          css: '[data-testid="page-container-footer-next"]',
        });
        await driver.clickElement({
          text: 'Connect',
          tag: 'button',
          css: '[data-testid="page-container-footer-next"]',
        });
        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);

        //
        // TODO: Somehow, some way, open the popup
        //
        driver.navigate(PAGES.POPUP, 'tab=0');

        // Create and switch to account 2
        await driver.waitForSelector('[data-testid="account-menu-icon"]');
        await driver.clickElement('[data-testid="account-menu-icon"]');
        await driver.clickElement(
          '[data-testid="multichain-account-menu-popover-action-button"]',
        );
        await driver.clickElement(
          '[data-testid="multichain-account-menu-popover-add-account"]',
        );
        await driver.clickElement({ text: 'Create', tag: 'button' });

        // Ensure toast is displaying and connects account when clicked
        const connectSelector = {
          text: 'Connect account',
          tag: '.toasts-container button',
        };
        await driver.waitForSelector(connectSelector);
        await driver.clickElement(connectSelector);

        // Ensure the account tooltip is displaying and correct connection color
        await driver.waitForSelector(
          'div[data-original-title="Account 2 connected"]',
        );
        await driver.waitForSelector(
          '.multichain-connected-site-menu__badge.mm-box--background-color-success-default',
        );

        // TEMPORARY: KEEP TEST OPEN
        await driver.clickElement('#xx');
      },
    );
  });
});

```

File: ui/components/multichain/pages/connections/components/no-connections.test.tsx [hunk 1]
```diff
@@ -1,16 +1,13 @@
 import React from 'react';
 import { renderWithProvider } from '../../../../../../test/jest';
-import {NoConnectionContent} from './no-connection';
+import { NoConnectionContent } from './no-connection';
```
Comment (NidhiKJha) [inline ui/components/multichain/pages/connections/components/no-connections.test.tsx:3]: I think this file shouldn't be included in this PR

File: ui/components/multichain/toast/toast.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  startAdornment: React.ReactNode | React.ReactNode[];
+  text: string;
+  actionText: string;
+  onActionClick: () => void;
+  onClose: () => void;
+}) => {
+  const { theme } = document.documentElement.dataset;
+
+  return (
+    <BannerBase
```
Comment (NidhiKJha) [inline ui/components/multichain/toast/toast.tsx:28]: We need to pass a className prop here

**Review by NidhiKJha — COMMENTED** [review]
For the already connected accounts, if I switch the account. I see the toast. But if I disconnect all my accounts and go to the dapp. It's not able to make a request to MetaMask to connect with it

https://github.com/MetaMask/metamask-extension/assets/39872794/038367f4-a253-4daf-95db-47ed7e2c1016

.

File: ui/components/multichain/toast/toast.tsx (hunk 2 above)
Comment (darkwing) [reply ui/components/multichain/toast/toast.tsx:28]: We could but we aren't using it anywhere.

**Comment by darkwing** [comment]
> For the already connected accounts, if I switch the account. I see the toast. But if I disconnect all my accounts and go to the dapp. It's not able to make a request to MetaMask to connect with it

We should investigate that as a separate issue.  The connections are working properly, which is what this toast is doing.

**Review by vthomas13 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]
LGTM
